daemon/main.go

```diff
@@ -12,362 +12,140 @@
-	}
+// RootCmd represents the base command when called without any subcommands
+var RootCmd = &cobra.Command{
+	Use:   "daemon",
+	Short: "Run cilium daemon",
+	Run: func(cmd *cobra.Command, args []string) {
+		initEnv()
+		runDaemon()
+	},
 }
 
-func statusDaemon(ctx *cli.Context) {
-	initClient(ctx)
-
-	if resp, err := client.Daemon.GetHealthz(nil); err != nil {
-		fmt.Fprintf(os.Stderr, "Error: Unable to reach out daemon: %s\n", err)
-		os.Exit(1)
-	} else {
-		sr := resp.Payload
-		w := tabwriter.NewWriter(os.Stdout, 2, 0, 3, ' ', 0)
-		if sr.Kvstore != nil {
-			fmt.Fprintf(w, "KVStore:\t%s\n", sr.Kvstore.State)
-		}
-		if sr.ContainerRuntime != nil {
-			fmt.Fprintf(w, "ContainerRuntime:\t%s\n", sr.ContainerRuntime.State)
-		}
-		if sr.Kubernetes != nil {
-			fmt.Fprintf(w, "Kubernetes:\t%s\n", sr.Kubernetes.State)
-		}
-		if sr.Cilium != nil {
-			fmt.Fprintf(w, "Cilium:\t%s\n", sr.Cilium.State)
-		}
-
-		if sr.IPAM != nil {
-			fmt.Printf("Allocated IPv4 addresses:\n")
-			for _, ipv4 := range sr.IPAM.IPV4 {
-				fmt.Printf(" %s\n", ipv4)
-
-			}
-			fmt.Printf("Allocated IPv6 addresses:\n")
-			for _, ipv6 := range sr.IPAM.IPV6 {
-				fmt.Printf(" %s\n", ipv6)
-			}
-		}
-
-		w.Flush()
-
-		if sr.Cilium != nil && sr.Cilium.State != models.StatusStateOk {
-			os.Exit(1)
-		} else {
-			os.Exit(0)
-		}
+func main() {
+	if err := RootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(-1)
 	}
-
 }
 
-func dumpConfig(Opts map[string]string) {
-	opts := []string{}
-	for k := range Opts {
-		opts = append(opts, k)
-	}
-	sort.Strings(opts)
-
-	for _, k := range opts {
-		text := common.Green("Enabled")
-
-		if Opts[k] == "" {
-			text = common.Red("Disabled")
-		}
-
-		fmt.Printf("%-24s %s\n", k, text)
-	}
+func init() {
+	cobra.OnInitialize(initConfig)
+	flags := RootCmd.Flags()
+	flags.StringVar(&cfgFile, "config", "", "config file (default is $HOME/ciliumd.yaml)")
+	flags.BoolP("debug", "D", false, "Enable debug messages")
+	flags.StringVar(&consulAddr, "consul", "", "Consul agent address [127.0.0.1:8500]")
+	flags.StringVarP(&config.Device, "device", "d", "undefined", "Device to snoop on")
+	flags.BoolVar(&disableConntrack, "disable-conntrack", false, "Disable connection tracking")
+	flags.BoolVar(&disablePolicy, "disable-policy", false, "Disable policy enforcement")
+	flags.StringVarP(&config.DockerEndpoint, "docker", "e", "unix:///var/run/docker.sock",
+		"Register a listener for docker events on the given endpoint")
+	flags.StringSliceVar(&etcdAddr, "etcd", []string{}, "Etcd agent address [http://127.0.0.1:2379]")
+	flags.StringVar(&config.EtcdCfgPath, "etcd-config-path", "", "Absolute path to the etcd configuration file")
+	flags.BoolVar(&enableTracing, "enable-tracing", false, "Enable tracing while determining policy")
+	flags.StringVar(&nat46prefix, "nat46-range", addressing.DefaultNAT46Prefix,
+		"IPv6 prefix to map IPv4 addresses to")
+	flags.StringVar(&config.K8sEndpoint, "k8s-api", "", "Kubernetes api address server")
```

Comment:
I'm fine with keeping it as `k8s-api-server`. I went back and forth on this one. I'll change it back.